client/consul_test.go

```diff
@@ -278,3 +278,57 @@ func TestConsul_ModifyCheck(t *testing.T) {
 		t.Fatalf("Expected number of check registrations: %v, Actual: %v", 2, apiClient.checkRegisterCallCount)
 	}
 }
+
+func TestConsul_FilterNomadServicesAndChecks(t *testing.T) {
+	c := newConsulService()
+	srvs := map[string]*consul.AgentService{
+		"foo-bar": {
+			ID:      "foo-bar",
+			Service: "http-frontend",
+			Tags:    []string{"global"},
+			Port:    8080,
+			Address: "10.10.1.11",
+		},
+		"nomad-2121212": {
+			ID:      "nomad-2121212",
+			Service: "identity-service",
+			Tags:    []string{"global"},
+			Port:    8080,
+			Address: "10.10.1.11",
+		},
+	}
+
```

Comment:
The best practice is to create a new expected map and then use reflect.DeepEquals:
```
exp := map[string]*consul.AgentService{
   "nomad-2121212": {
			ID:      "nomad-2121212",
			Service: "identity-service",
			Tags:    []string{"global"},
			Port:    8080,
		Address: "10.10.1.11",
		},

   act :=  c.filterConsulServices(serves)
   if !reflect.DeepEquals(act, exp) {
       t.Fatalf(...)
   }
}
```